### PR TITLE
`seal_reentrant_count` returns contract reentrant count

### PR DESCRIPTION
**Pull request by Artemka374**
This PR addresses [#11082](https://github.com/paritytech/substrate/issues/11082).

### CODE REVIEW

File: frame/contracts/src/tests.rs [hunk 1]
```diff
@@ -4383,3 +4383,114 @@ fn delegate_call_indeterministic_code() {
...
+			Determinism::Deterministic,
+		)
+		.result
+		.unwrap();
+	});
+}
+
+#[test]
+#[cfg(feature = "unstable-interface")]
+fn account_entrance_count_works() {
```
Comment (athei) [inline frame/contracts/src/tests.rs:4463]: This test is insufficient:

- It tests only for its own address: Please additionally test for the address of **another** contract on the call stack.
- This only test for the value `1`. Please also test at least for `0`, too.

**Review by athei — CHANGES_REQUESTED** [review]
Please address comments.

**Comment by Artemka374** [comment]
fixed, please check again

**Review by agryaznov — APPROVED** [review]
LGTM

File: frame/contracts/src/tests.rs (hunk 1 above)
Comment (athei) [reply frame/contracts/src/tests.rs]: The test still tests the trivial case where nothing is on the call stack. But let's not be too strict here.

**Review by athei — APPROVED** [review]

**Pull request merged**